pfuture.el

```diff
@@ -175,35 +225,58 @@ details see documentation of `accept-process-output'."
      process timeout nil just-this-one))
   (process-get process 'result))
 
+(cl-macrolet
+    ((define-getter (name doc variable )
+       `(define-inline ,name (process)
+          ,doc
+          (declare (side-effect-free t))
+          (inline-letevals (process)
+            (inline-quote
+             (process-get ,',process ',variable))))))
+  (define-getter pfuture-result "Return the output of a pfuture PROCESS." stdout)
+  (define-getter pfuture-stderr "Return the error output of a pfuture PROCESS." stderr))
+
 (defun pfuture-await-to-finish (process)
   "Keep reading the output of PROCESS until it is done.
 Same as `pfuture-await', but will keep reading (and blocking) so long as the
 process is *alive*.
 
 If the process never quits this method will block forever. Use with caution!"
+  ;; If the sentinel hasn't run, disable it. We are going to delete
+  ;; the stderr process here.
+  (set-process-sentinel process nil)
   (let (inhibit-quit)
-    (accept-process-output process nil nil t)
-    (while (process-live-p process)
-      (accept-process-output process nil nil t)))
-  (process-get process 'result))
-
-(define-inline pfuture-result (process)
-  "Return the output of a pfuture PROCESS."
-  (declare (side-effect-free t))
-  (inline-letevals (process)
-    (inline-quote
-     (process-get ,process 'result))))
-
-(defun pfuture--append-output (process msg)
-  "Append PROCESS' MSG to the already saved output."
-  (process-put process 'result (concat (process-get process 'result) msg)))
+    (while (accept-process-output process)))
```

Comment:
`accept-process-output` returns `t` until the process exits, so a check for liveness is unnecessary. Having `just-this-one` be `t` caused other processes not to be cleaned up during the wait, so e.g.

```
   (dotimes (i 1000)
     (pfuture-await-to-finish (pfuture-new "ls")))
```

used to fail with too many open pipes.